text/068-coqide-split.md

```diff
@@ -0,0 +1,102 @@
+
+The drawbacks of CoqIDE and coqidetop are that:
+
+- maintaining a standalone IDE just for Coq means that a lot of standard work has to be redone and lots of standard IDE features are not implemented,
+- the XML protocol is entirely custom and suited for the CoqIDE implementation, which means that any IDE wanting to interface with Coq through this protocol has to reimplement a lot of things, and that it won't always be well-suited to the model that the IDE expects.
+
+Recently, the Coq team has changed its strategy, by focusing instead on developing support for the [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) (LSP) with minimal extensions, which should bring the possibility to develop Coq support for any standard IDE (starting with VS Code). Meanwhile, CoqIDE, coqidetop and the STM are considered outdated technology that the Coq team would like to avoid putting too much work into maintaining.
+
+Because *some users still prefer a standalone IDE for Coq*, and some are *motivated to contribute to CoqIDE*, we want to give the opportunity to the community to take over CoqIDE maintenance by splitting out the CoqIDE source code from Coq's repository, on the condition that volunteer maintainers are found.
+
+## Detailed design
+
+### Short-term actions
+
+Upon acceptance of this CEP, we will send an open call for volunteer maintainers. The planned text of the call is available below, in an appendix to the CEP's main text.
+
+In the case where volunteer maintainer(s) are found:
+
+1. The code for CoqIDE (the UI part) will be split into a separate repository (the code for coqidetop and the STM will remain in the Coq repository for as long as there are other major users besides CoqIDE that depend on them). The repository will temporarily be created in the Coq organization.
+2. Any issue in the Coq issue tracker pertaining to CoqIDE will be moved to this new repository.
+3. The repository will be transferred to the [Coq-community organization](https://github.com/coq-community/manifesto).
+4. The volunteer maintainer(s) will be given admin access to the repository.
+5. CoqIDE will be added in Coq's Continuous Integration (CI), but there is no commitment from the Coq team that it will get to stay there indefinitely.
+6. CoqIDE will keep being distributed as part of the [Coq Platform](https://github.com/coq/platform), for as long as it respects the inclusion criteria (timely release of compatible CoqIDE versions with new Coq releases), but will no longer be considered an official project by the Coq team.
+
+The new CoqIDE maintainers will receive support from both the Coq team (for as long as CoqIDE is in Coq's CI, more about this in the next subsection) and from the Coq-community organization owners (to set up solutions for CI, deploying documentation, protecting branches, etc.), but they will have complete freedom to define:
+
+- what features / changes to include,
```

Comment:
That seems like a tricky point, imagine new CoqIDE maintainers decide to drop the debugger (which IMHO needs reimplementing), what would that mean? This is also relevant to point 1, as keeping the STM in Coq codebase does greatly hamper development for other IDEs and sustract quite a lot of eng resources to maintain it.

Reply:
If the debugger isn't available in vsCoq2 at that time, I think the users of the debugger would be up in arms.